twake/backend/node/src/services/messages/web/controllers/views.ts

```diff
@@ -1,26 +1,65 @@
 import { FastifyReply, FastifyRequest } from "fastify";
-import { CrudController } from "../../../../core/platform/services/webserver/types";
 import { MessageServiceAPI } from "../../api";
-import {
-  ResourceCreateResponse,
-  ResourceDeleteResponse,
-  ResourceGetResponse,
-  ResourceListResponse,
-} from "../../../../utils/types";
+import { ResourceListResponse } from "../../../../utils/types";
 import { Message } from "../../entities/messages";
 import { handleError } from "../../../../utils/handleError";
 import { ListResult, Pagination } from "../../../../core/platform/framework/api/crud-service";
 import {
   ChannelViewExecutionContext,
   MessageViewListOptions,
-  PaginationQueryParameters,
   MessageWithReplies,
+  PaginationQueryParameters,
 } from "../../types";
 import { keyBy } from "lodash";
 import { RealtimeServiceAPI } from "../../../../core/platform/services/realtime/api";
+import ChannelServiceAPI from "../../../channels/provider";
+import WorkspaceServicesAPI from "../../../workspaces/api";
+
+class SimpleChannelsCache {
+  private readonly data: { [key: string]: { [key: string]: boolean } };
+  private readonly expiration: { [key: string]: number };
+  static now = () => new Date().getTime();
+
+  constructor(protected minutesTTL: number) {
+    this.data = {};
+    this.expiration = {};
+    setInterval(() => {
+      const now = SimpleChannelsCache.now();
+      Object.keys(this.expiration).forEach(key => {
+        if (now > this.expiration[key]) {
+          delete this.expiration[key];
+          delete this.data[key];
+        }
+      });
+    }, minutesTTL * 1000);
+  }
+
+  async get(key: string, setIfMissing: () => Promise<{ [key: string]: boolean }> = undefined) {
+    const exp = this.expiration[key];
+    if (!exp || SimpleChannelsCache.now() > exp) {
+      if (!setIfMissing) {
+        return null;
+      }
+      this.set(key, await setIfMissing());
+    }
+    return this.data[key];
+  }
+
+  set(key: string, value: { [key: string]: boolean }) {
+    this.expiration[key] = SimpleChannelsCache.now() + this.minutesTTL * 1000;
+    this.data[key] = value;
+  }
+}
+
+const simpleChannelsCache = new SimpleChannelsCache(5);
```

Comment:
Actually we already have a cache available `import NodeCache from "node-cache";` you can see it in action here: twake/backend/node/src/services/user/services/users/service.ts

Reply:
🤦‍♂️ … 
mine is pretty elegant too )
ok, I'll change it to node-cache